src/main/scala/mesosphere/marathon/core/volume/providers/Helpers.scala

```diff
@@ -23,21 +23,17 @@ protected[volume] abstract class AbstractPersistentVolumeProvider(
 protected[providers] object OptionSupport {
   import OptionLabelPatterns._
 
-  /** NamedOption represents a (named) configurable item type that provides validation rules */
-  abstract class NamedOption(
-      val namespace: String,
-      val name: String,
-      val required: Boolean,
-      val validValue: Validator[String]) {
-
-    def fullName: String = namespace + OptionNamespaceSeparator + name
-    def from(m: Map[String, String]): Option[String] = m.get(fullName)
+  def validIfDefined[T](implicit validator: Validator[T]): Validator[Option[T]] = new Validator[Option[T]] {
```

Comment:
/cc @kolloch any good place to put these? Or any other way to elegantly express the same?